packages/cozy-stack-client/src/getIconURL.spec.js

```diff
@@ -39,142 +41,192 @@ describe('get icon', () => {
+      }
+      const url = await getIconURL(stackClient, {
+        type: 'konnector',
+        slug: 'caissedepargne10'
       })
-    )
+      expect(global.URL.createObjectURL).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: 'image/svg+xml'
+        })
+      )
+      expect(url.data[0]).toBe('<svg></svg>')
+      resetCreateObjectURL()
+      await getIconURL(stackClient, defaultOpts)
+
+      expect(global.URL.createObjectURL).not.toHaveBeenCalled()
+    })
   })
 
-  it('should call the server the second time if the first time it failed', async () => {
-    responses['/registry/caissedepargne1/icon'] = {
-      ok: true,
-      blob: () => FakeBlob([svgData], { type: 'image/svg+xml' })
-    }
-    stackClient.fetch = jest.fn().mockRejectedValue('No connexion')
+  describe('when consuming app is not using oauth', () => {
+    beforeEach(() => {
+      stackClient.oauthOptions = undefined
+      stackClient.uri = 'http://cozy.tools:8080/anywhere/where/user/can/be'
+    })
 
-    await getIconUrlDefault(stackClient, defaultOpts)
-    expect(global.URL.createObjectURL).not.toHaveBeenCalled()
+    it('should not create object url as blob', async () => {
+      await getIconURL(stackClient, defaultOpts)
+      expect(global.URL.createObjectURL).not.toHaveBeenCalled()
+    })
 
-    stackClient.fetch = jest.fn().mockImplementation(fakeResp)
-    const url2 = await getIconUrlDefault(stackClient, defaultOpts)
-    expect(global.URL.createObjectURL).toHaveBeenCalledWith(
-      expect.objectContaining({
-        type: 'image/svg+xml'
-      })
-    )
-    expect(url2.data[0]).toBe('<svg></svg>')
-  })
+    it('should return url from appData, links, icon if defined', async () => {
+      defaultOpts.appData = { links: { icon: '/path/to/icon' } }
+      const url = await getIconURL(stackClient, defaultOpts)
+      expect(url).toEqual('http://cozy.tools:8080/path/to/icon')
+    })
 
-  it('should not call create createObjectURL since it should be memoized', async () => {
-    /**
-     * Create a new response / request to be sure to not having memoized it already
-     */
-    responses['/konnectors/caissedepargne10/icon'] = {
-      ok: true,
-      blob: () => FakeBlob([svgData], { type: 'image/svg+xml' })
-    }
-    const url = await getIconUrlDefault(stackClient, {
-      type: 'konnector',
-      slug: 'caissedepargne10'
-    })
-    expect(global.URL.createObjectURL).toHaveBeenCalledWith(
-      expect.objectContaining({
-        type: 'image/svg+xml'
-      })
-    )
-    expect(url.data[0]).toBe('<svg></svg>')
-    resetcreateObjectURL()
-    await getIconUrlDefault(stackClient, defaultOpts)
+    it('should return url from appData if appData is a string', async () => {
+      defaultOpts.appData = 'appData-that-is-a-string'
+      const url = await getIconURL(stackClient, defaultOpts)
+      expect(url).toEqual(
+        'http://cozy.tools:8080/registry/appData-that-is-a-string/icon'
```

Comment:
@Crash-- : I wonder if this appData that can be string is possible

Reply:
Nein. I don't think so.

Reply:
this code + test have been removed